geonode/qgis_server/helpers.py

```diff
@@ -421,6 +425,241 @@ def legend_url(layer, layertitle=False, internal=True):
     return url
 
 
+def wms_get_capabilities_url(layer=None, internal=True):
+    """Construct WMS GetCapabilities request.
+
+    :param layer: Layer to inspect
+    :type layer: Layer
+
+    :param internal: Flag to switch between public url and internal url.
+        Public url will be served by Django Geonode (proxified).
+    :type internal: bool
+
+    :return: QGIS Server request url
+    :rtype: str
+    """
+    try:
+        qgis_layer = QGISServerLayer.objects.get(layer=layer)
+    except QGISServerLayer.DoesNotExist:
```

Comment:
I tried to use get_object_or_404 only for view function.
These helpers might be used somewhere else other than in the view function directly.
So I think it is better to keep the error context (rather than a http context 404 error).